cfme/infrastructure/virtual_machines.py

```diff
@@ -62,100 +65,30 @@
-        vm_templates_tree(
-            "All VMs & Templates", ctx["provider_name"], ctx["datacenter_name"], ctx["vm_name"])
-
-    def vm_templates_archived_branch(ctx):
-        vm_templates_tree("All VMs & Templates", "<Archived>")
-
-    def infra_archive_obj(ctx):
-        vm_templates_tree("All VMs & Templates", "<Archived>", ctx["archive_name"])
-
-    def vm_templates_orphaned_branch(ctx):
-        vm_templates_tree("All VMs & Templates", "<Orphaned>")
+match_page = partial(match_location, controller='vm_infra', title='Virtual Machines')
 
-    def infra_orphan_obj(ctx):
-        vm_templates_tree("All VMs & Templates", "<Orphaned>", ctx["orphan_name"])
 
-    class infra_vms:
-        def navigate(_):
-            vms_tree("All VMs")
+def nav_to_vms():
+    from cfme.web_ui.menu import nav
+    nav._nav_to_fn('Compute', 'Infrastructure', '/vm_infra/explorer')(None)
 
-        def infra_vm_by_name(ctx):
-            sel.click(ctx['vm'].find_quadicon(do_not_navigate=True))
 
-    def infra_vms_filter_folder(ctx):
-        vms_tree("All VMs", ctx["folder_name"])
-
-    def infra_vms_filter(ctx):
-        vms_tree("All VMs", ctx["folder_name"], ctx["filter_name"])
-
-    def infra_templates(_):
-        templates_tree("All Templates")
-        toolbar.select('Grid View')
-
-    def infra_templates_filter_folder(ctx):
-        templates_tree("All Templates", ctx["folder_name"])
-
-    def infra_templates_filter(ctx):
-        templates_tree("All Templates", ctx["folder_name"], ctx["filter_name"])
-
-
-class Common(object):
-    """Stuff shared for bot hVM and Template."""
-    def on_details(self, force=False):
-        """A function to determine if the browser is already on the proper vm details page.
-        """
-        title = '{} "{}"'.format(
-            "VM and Instance" if self.is_vm else "VM Template and Image", self.name)
-        present_title = summary_title()
-        if present_title is None or title not in present_title:
-            if not force:
-                return False
-            else:
-                self.load_details()
-                return True
-
-        pattern = r'("[A-Za-z0-9_\./\\-]*")'
-        m = re.search(pattern, present_title)
+def reset_page():
+    tb.select("Grid View")
+    if paginator.page_controls_exist():
+        # paginator.results_per_page(1000)
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
 
-        if not force:
-            return self.name == m.group().replace('"', '')
-        else:
-            if self.name != m.group().replace('"', ''):
-                self._load_details()
-            else:
-                return False
 
-    @property
-    def genealogy(self):
-        return Genealogy(self)
+drift_table = CheckboxTable({
+    version.LOWEST: "//table[@class='style3']",
+    "5.4": "//th[normalize-space(.)='Timestamp']/ancestor::table[1]"
+})
 
 
 @BaseVM.register_for_provider_type("infra")
-class Vm(BaseVM, Common):
+class Vm(BaseVM):
```

Comment:
this class shared two common methods for both Template and Vm but one of them became unnecessary after moving to navmazing.
second one was just simple property. it didn't make sense to keep such class only for that property.